src/docfx/restore/CredentialHandler.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    internal class CredentialHandler : DelegatingHandler
+    {
+        private readonly List<KeyValuePair<string, HttpConfig>> _credentials;
+
+        public CredentialHandler(List<KeyValuePair<string, HttpConfig>> credentials, HttpMessageHandler innerHandler)
+            : base(innerHandler)
+        {
+            _credentials = credentials;
+        }
+
+        public CredentialHandler Create(HttpMessageHandler innerHandler)
```

Comment:
>Create [](start = 33, length = 6)

this looks weird - why do we want to new a `CredentialHandler` with exiting credentials

Reply:
it is to create `credentialHandler` with a different `innerHandler`